app/abuser_utils.py

```diff
@@ -0,0 +1,272 @@
+        mailboxes: List[Mailbox] = Mailbox.filter_by(user_id=user.id).all()
+        bundle = {
+            "account_id": str(user.id),
+            "email": primary_email,
+            "user_created_at": user.created_at.isoformat() if user.created_at else None,
+            "aliases": [
+                {
+                    "address": alias.email.lower() if alias.email else None,
+                    "created_at": alias.created_at.isoformat()
+                    if alias.created_at
+                    else None,
+                }
+                for alias in aliases
+                if alias.email
+            ],
+            "mailboxes": [
+                {
+                    "address": mailbox.email.lower() if mailbox.email else None,
+                    "created_at": mailbox.created_at.isoformat()
+                    if mailbox.created_at
+                    else None,
+                }
+                for mailbox in mailboxes
+                if mailbox.email
+            ],
+        }
+        bundle_json_bytes = json.dumps(bundle, sort_keys=True).encode("utf-8")
+        k_bundle_random = AESGCM.generate_key(bit_length=256)
+        aesgcm_bundle_enc = AESGCM(k_bundle_random)
+        nonce_bundle = secrets.token_bytes(12)
+        aad_bytes = constants.AEAD_AAD_DATA.encode("utf-8")
+        encrypted_bundle_data = nonce_bundle + aesgcm_bundle_enc.encrypt(
+            nonce_bundle, bundle_json_bytes, aad_bytes
+        )
+        abuser_data_entry = AbuserData(
+            user_id=user.id, encrypted_bundle=encrypted_bundle_data
+        )
+
+        Session.add(abuser_data_entry)
+        Session.flush()
+
+        blob_id = abuser_data_entry.id
+        all_identifiers_raw = (
+            [primary_email]
+            + [a.email for a in aliases if a.email]
+            + [m.email for m in mailboxes if m.email]
+        )
+        seen_normalized_identifiers = set()
+        mac_key_bytes = config.MAC_KEY
+        master_key_bytes = config.MASTER_ENC_KEY
+
+        for raw_identifier_address in all_identifiers_raw:
+            if not raw_identifier_address:
+                continue
+
+            normalized_identifier = raw_identifier_address.lower()
+
+            if normalized_identifier in seen_normalized_identifiers:
+                continue
+
+            seen_normalized_identifiers.add(normalized_identifier)
+            identifier_hmac = hmac.new(
+                mac_key_bytes, normalized_identifier.encode("utf-8"), sha256
+            ).hexdigest()
+
+            k_identifier_derived = _derive_key_for_identifier(
+                master_key_bytes, normalized_identifier
+            )
+            aesgcm_key_enc = AESGCM(k_identifier_derived)
+            nonce_key_encryption = secrets.token_bytes(12)
+            encrypted_k_bundle_for_this_identifier = (
+                nonce_key_encryption
+                + aesgcm_key_enc.encrypt(
+                    nonce_key_encryption, k_bundle_random, aad_bytes
```

Comment:
```suggestion
                    nonce_key_encryption, k_bundle_random, f"{aad_bytes}.bundle"
```
So the tag is different for the bundle as from the keys